### PR TITLE
Consensus Chapter

### PR DESCRIPTION
**Pull request by Hoverbear**
Adds most of the Consensus Chapter. I need to write the Paxos bit still.

### CODE REVIEW

**Comment by siddontang** [comment]
please separate them by two PRs, one is for the real doc, one is for book.

**Comment by Hoverbear** [comment]
How about two commits? One the book edits, the other the generated content?

**Comment by siddontang** [comment]
@Hoverbear 

It is hard to review the PR if you introduce the HTML page. we can build a bot to help us generate automatically, not in this PR.

**Comment by Hoverbear** [comment]
@siddontang For now I think you can just review the one commit: https://github.com/tikv/deep-dive-tikv/pull/7/commits/e1b9d8c3a676dc5e3eaddae3072892c108793a2e

The second commit is all the rubbish it generates.

**Comment by siddontang** [comment]
@Hoverbear 

I think for most of the reviewers, they will click the `File Changes` and review your PR

image

But when they enter the page, they find many HTML which is hard to review, this is confusing.

So I insist on that you should use another PR to submit the HTML changes.

File: src/consensus-algorithm/paxos.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# Paxos
```
Comment (Hoverbear) [inline src/consensus-algorithm/paxos.md:1]: TODO for the moment. I will fill this soon.

File: src/consensus-algorithm/introduction.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Consensus Algormithm
+
+When building a distributed system one principal goal is often to build in fault-tolerance. That is, if one particular node in a network goes down, or if there is a network partition, the entire cluster does not fall over. The cluster of nodes taking part in a distributed consensus protocol must come to agreement regarding values, and once that decision is reached, that choice is final.
+
+Distributed Consensus Algorithms often take the form of a replicated state machine and log. Each state machine accepts inputs from its log, and represents the value(s) to be replicated, for example, a hash table. They allow a collection of machines to work as a coherent group that can survive the failures of some of its members.
```
Comment (brson) [inline src/consensus-algorithm/introduction.md]: The sentence "Each state machine accepts inputs from its log, and represents the value(s) to be replicated, for example, a hash table" isn't clear to me. Is the hash table the value to be replicated, or is the hash table how it's storing the value to be replicated. What does it mean to 'represent' a value? What's the relationship between the 'input from its log', and 'the value to be replicated'?

Can you clarify this sentence?

**Review by brson — COMMENTED** [review]
@Hoverbear I've reviewed this chapter and think you should rebase and merge it right away, though if you could clarify the meaning of that sentence I pointed out that would be even better. lgtm

**Comment by brson** [comment]
Once this is merged, I have another patch of copyediting for this chapter: https://github.com/brson/deep-dive-tikv/tree/copyedit-consensus

File: src/consensus-algorithm/introduction.md (hunk 2 above)
Comment (Hoverbear) [reply src/consensus-algorithm/introduction.md]: Yes, definitely.

**Comment by Hoverbear** [comment]
@brson PTAL, should be good to merge. Includes your copy edits.

**Comment by brson** [comment]
lgtm Thanks for the quick turnaround.

**Pull request merged**